### PR TITLE
Filter query count first

### PR DESCRIPTION
**Pull request by adube**
This PR introduces the concept of doing "count" requests before WFS GetFeature requests, as supported in the current query service.

The `ngeo.Querent` is responsible of doing the requests.  The `ngeo.MapQuerent` reads what's returned and use it in the result.

## Todo

 * [x] wait for #2223 to be merged first
 * [x] wait for #2239 to be merged first
 * [ ] review (see last commits (those that have 'count'))
 * [ ] fix issues reported by Travis

### CODE REVIEW

**Comment by adube** [comment]
Rebased.

File: contribs/gmf/examples/displayquerygrid.js [hunk 1]
```diff
@@ -9,7 +9,9 @@ goog.require('gmf.layertreeDirective');
 /** @suppress {extraRequire} */
 goog.require('gmf.mapDirective');
 /** @suppress {extraRequire} */
-goog.require('ngeo.proj.EPSG21781');
+goog.require('ngeo.proj.EPSG21781')
+;/** @suppress {extraRequire} */
```
Comment (sbrunner) [inline contribs/gmf/examples/displayquerygrid.js]: missplaced `;`

File: src/services/syncDataSourcesMap.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  }
+
+  /**
+   * Called when the resolution of the map view changes. Synchronize the
+   * datasources to current resolution of the view.
+   * @param {ol.ObjectEvent} evt Event.
+   * @private
+   */
+  handleViewResolutionChange_(evt) {
+    const view = /** @type {ol.View} */ (evt.target);
```
Comment (sbrunner) [inline src/services/syncDataSourcesMap.js]: avoid type cast...

**Comment by adube** [comment]
Any idea why Travis fails?

**Comment by sbrunner** [comment]
I think that's explicit:
https://travis-ci.org/camptocamp/ngeo/jobs/195975763#L973-L980
I'm having a look for the last one

File: options/ngeox.js [hunk 3]
```diff
@@ -294,6 +334,159 @@ ngeox.DataSourceOptions.prototype.wmtsUrl;
...
+ngeox.QuerentResult;
+
+
+/**
+ * The definition of a result item returned by the querent service.
+ *
+ * @typedef {{
+ *     features: (Array.<ol.Feature>),
+ *     tooManyFeatures: (boolean|undefined),
+ *     totalFeatureCount: (number|undefined),
```
Comment (sbrunner) [inline options/ngeox.js]: I think that' there is an extra coma here?

Comment (adube) [inline options/ngeox.js]: Indeed.  Why isn't `make dist` reporting this?

**Comment by fredj** [comment]
looks good to me too, thanks @adube

**Comment by adube** [comment]
Thanks!

**Pull request merged**